### PR TITLE
Mantine Button

### PR DESCRIPTION
**Pull request by ranquild**
Changes:
- Exports `Button` and add style overrides
- Adds focus ring styles

How to verify:
- Run `yarn storybook`
- Open http://localhost:6006/?path=/story/buttons-button--default-story
- Make sure it works according to the figma file https://www.figma.com/file/Ey1rOyIxRHpmRvE9XrGyop/Buttons-%2F-Button?type=design&node-id=1-96&mode=design&t=DaIKPKduzsl5PHpd-0
- For focus ring, click somewhere in "canvas" and press `Tab`

Screenshot 2023-08-16 at 15 33 06

### CODE REVIEW

File: frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        rightIcon: {
+          marginLeft: theme.spacing.sm,
+        },
+      };
+    },
+  },
+});
+
+const getButtonVariantStyles = (theme: MantineTheme, variant?: string) => {
+  const hoverColor = theme.fn.rgba(theme.fn.primaryColor(), 0.88);
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx:79]: This is how currently we compute the hover color for the existing `Button`. Please note that it doesn't match `Hover` color in Figma and this makes buttons look somewhat inconsistent.

**Comment by dhuniverse** [comment]
- Are you going to implement other variations as mocked up in Figma? E.g, Centered with full width, pill shape etc, compact size.

- Might be better to use real UI examples for the button labels. E.g., Summarize for Default button, "Convert this question to SQL" for Primary button, "+ New" for icon button etc.

- Will provide doc text in Figma: https://www.figma.com/file/Ey1rOyIxRHpmRvE9XrGyop/Buttons-%2F-Button?type=design&node-id=418%3A2983&mode=design&t=sKgLQYexSUp0MQMB-1

**Comment by ranquild** [comment]
@dhuniverse added docs, all screens from Figma, and used realistic button labels

File: frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+          lineHeight: "1rem",
+          color: styles.color,
+          borderColor: styles.borderColor,
+
+          "&:disabled": {
+            color: styles.disabledColor,
+            borderColor: styles.disabledBorderColor,
+            backgroundColor: styles.disabledBackgroundColor,
+          },
+          "&[data-loading]": {
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx:38]: `loading` is not available in `params`

File: frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            [`& .${getStylesRef("icon")}`]: {
+              color: styles.loaderColor,
+            },
+          },
+          "&:hover": {
+            color: styles.hoverColor,
+            borderColor: styles.hoverBorderColor,
+            backgroundColor: styles.hoverBackgroundColor,
+          },
+          [`&:has(.${getStylesRef("label")}:empty)`]: {
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/buttons/Button/Button.styled.tsx:52]: `children` is also not available in params, do I had to do this to change styles when there is no button label

File: frontend/src/metabase/ui/components/overlays/Menu/Menu.stories.mdx [hunk 4]
```diff
@@ -121,9 +121,9 @@ export const Default = args => (
   <Story name="Default">{Default}</Story>
 </Canvas>
 
-### With a right section
+### Right section
 
-export const WithRightSection = args => (
+export const RightSection = args => (
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/overlays/Menu/Menu.stories.mdx:126]: For consistency with `Button` stories

**Review by oisincoveney — COMMENTED** [review]
Looks good, just one thing I saw with the `IconOnly` story. Maybe it's intended but it looks like the icon is just slightly offset to the left - I expected the icon to be in the center

image

I saw when inspecting the span for `css-Button-icon` that there's a weird `margin-right:0.25rem` that, when removed, does help a little bit

**Comment by ranquild** [comment]
@oisincoveney could you double check you have this issue on the latest version? If you do, could you post the screenshot?

**Comment by oisincoveney** [comment]
@ranquild Yeah sorry about that 😅 I really like this set up with all of the buttons in one place

But on the topic of the off-center icon, it's still happening. Here's a screenshot of the right-most buttons from my view 

Screenshot 2023-08-16 at 5 31 15 PM


I had a look at the circle icon in Figma to see if I could have some data to back this up - I set up a square around the button and put a line halfway down the middle, and this is what I saw (I can't seem to take a screenshot while showing the measurements due to conflicting shortcuts). 

image

**Comment by ranquild** [comment]
@oisincoveney it seems like the icon itself is not centered. There is no extra margin or padding

Screenshot 2023-08-16 at 17 43 27

**Comment by oisincoveney** [comment]
@ranquild Ah I see, in that case I have no issues on my side, it all looks really great

**Review by oisincoveney — APPROVED** [review]

**Review by kdoh — APPROVED** [review]

**Pull request merged**